music21/note.py

```diff
@@ -1671,6 +1690,32 @@ def displayPitch(self) -> pitch.Pitch:
         p.octave = self.displayOctave
         return p
 
+    @property
+    def midi(self) -> int:
+        '''
+        Returns the MIDI pitch corresponding to the stored display pitch.
+
+        >>> unp = note.Unpitched('D4')
+        >>> unp.midi
+        62
+        >>> unp2 = note.Unpitched()
+        >>> unp2.midi
+        60
+        '''
+        return self.displayPitch().midi
```

Comment:
Not sure I want this in yet -- is there a reason it needs to be here?   I think people would expect the MIDI not to be dependent on the displayPitch but on the storedInstrument's percussion MIDI mapping.

Reply:
I need a diatonic note number to sort the notes from musicxml documents here:

https://github.com/cuthbertLab/music21/blob/c69968d44213a1915d1e578f7509f3a7642c0e3b/music21/musicxml/xmlToM21.py#L2680-L2682

but I should probably just implement `diatonicNoteNum` instead.

Reply:
Nope! I see what you mean, I was accessing the displayPitch() here directly anyway, so I don't need this implemented here.